Sources/Core/SoftwareBreakpointManager.cpp

```diff
@@ -124,4 +124,31 @@ bool SoftwareBreakpointManager::fillStopInfo(Target::Thread *thread,
   stopInfo.reason = StopInfo::kReasonBreakpoint;
   return true;
 }
+
+void SoftwareBreakpointManager::insertStashedInsns(Address const &start,
+                                                   size_t length,
+                                                   ByteVector &data) {
+  Address end = start + length;
+  for (auto const &insn : _insns) {
+    Address insn_start = insn.first;
+    Address insn_end = insn.first + insn.second.size();
+
+    // no overlap between region and instruction
+    if (end < insn_start || start >= insn_end) {
+      continue;
+    }
+
+    // overlap begins at beginning of instruction
+    if (start <= insn_start) {
+      uint64_t offset = insn_start - start;
+      uint64_t size = std::min((uint64_t)insn.second.size(), end - insn_start);
+      memcpy(&data[offset], &insn.second[0], size);
+    } else {
+      DS2ASSERT(start < insn_end);
+      uint64_t offset = insn_end - start;
```

Comment:
I'm pretty sure this should be `start - insn_start`, since you want the index into the instruction to start writing?

For example, I think it will fail here:
```
insn_start := 0
start := 1
insn_end := 4
```
The instruction is 4 bytes long. Your code calculates the offset to be 4 -  1 = 3, which means you only write the last byte in the instruction when you wanted to copy the last 3 bytes.

I might be wrong, so you should check my understanding of this.